internal/core/api.rs

```diff
@@ -741,6 +742,64 @@ pub enum EventLoopError {
     NoEventLoopProvider,
 }
 
+/// The platform encountered a fatal error.
+///
+/// Either it couldn't be initialized, or some problem happened with the display server
+///
+/// This can be constructed from a `String`:
+/// ```rust
+/// use slint::platform::PlatformError;
+/// PlatformError::from(format!("Could not load resource {}", 1234));
+/// ```
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum PlatformError {
+    /// There was no default platform selected, and no platform was initialized
+    ///
+    /// Either a platform must be selected trough the `backend-*` cargo features flags,
+    /// or [`platform::set_platform()`](crate::platform::set_platform) must be called
+    /// before running the event loop
+    NoPlatform,
+    /// The Slint Platform does not provide an event loop.
+    ///
+    /// The [`Platform::run_event_loop`](crate::platform::Platform::run_event_loop)
+    /// is not implemented for the current platform
+    NoEventLoopProvider,
+
+    /// There is already a platform set from another thread another thread
+    SetPlatformError(crate::platform::SetPlatformError),
+
+    /// Another platform-specific error occurred
+    Other(String),
+}
+
+impl core::fmt::Display for PlatformError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            PlatformError::NoPlatform => f.write_str(
+                "There was no default Slint platform selected, and no Slint platform was initialized",
```

Comment:
Just formatting to use "XXX was selected, XXX was initialized" in both parts of the sentence.

```suggestion
                "No default Slint platform was selected, and no Slint platform was initialized",
```